drivers/shared/executor/executor_linux.go

```diff
@@ -188,7 +213,20 @@ func (l *LibcontainerExecutor) Launch(command *ExecCommand) (*ProcessState, erro
 	// be multiplexed
 	l.userProcExited = make(chan interface{})
 
-	go l.wait()
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		defer cancel()
+
+		go l.wait()
+
+		signal := l.catchSignals(ctx)
```

Comment:
Once we call `go l.wait()` we're continuing execution to the signal handler, but this thread could at this point jump to the `go l.wait()` goroutine and have it return before our signal handler is set up. I think we want to set up the signal handler first and then wait?